<a href="https://colab.research.google.com/github/revaanwarr/pemrograman-lanjut/blob/main/Phyton_Reva_ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**🖥️👩🏻‍💻PRAKTIKUM PEMROGRAMAN LANJUT🔥**
##**Pertemuan (1) PENDAHULUAN PHYTON - Senin, 14 Oktober 2024**

👤 Nama: Reva Anwar / NRP: 2C2230003
 / Kelas: III/A Sains Data

💬Deskripsi Pembahasan Materi:

1.1.  Beberapa karakteristik dari bahasa pemrograman Python:

- Python bersifat case-sensitive, artinya perbedaan huruf besar dan huruf kecil menyebabkan
perbedaan makna sintaks.

- Python sangat memerhatikan indentasi dan pergantian baris. Kesalahan indentasi dapat
menyebabkan gagal compile hingga kesalahan program.

- Variabel di python bersifat implisit dan dinamis. Artinya, sebuah variabel tidak perlu dideskripsikan
tipe datanya.


1.2.Macam - macam tipe data‍

- Bool(Boolean) : True atau False

- Int(Bilangan bulat) : Seluruh bilangan bulat

- Float(Bilangan real : Seluruh bilangan real

- String(Teks) : Kumpulan karakter


#🤸🏻‍♀️ EKSPLORASI PRIBADI DI KELAS ✨

In [ ]:

X = 101
Y= 4

X//Y
X/Y

print(X//Y)
print(X/Y)


25
25.25


- Interpretasi coding:

- Interpretasi Hasil:

#🧠 LATIHAN SOAL DI KELAS ⏱️

Problem (1) Ganjil-Genap

In [ ]:
# Meminta input dari pengguna
N = int(input("Masukkan nilai N: "))

# Memeriksa apakah N adalah positif, negatif, atau nol
if N > 0:
    if N % 2 == 0:
        print(f"{N} adalah bilangan positif genap")
    else:
        print(f"{N} adalah bilangan positif ganjil")
elif N < 0:
    print(f"{N} adalah bilangan negatif")
else:
    print("N adalah nol")

Masukkan nilai N: 2
2 adalah bilangan positif genap


- Interpretasi coding:

Meminta penggna masukkan nilai N, memeriksa aakah bilangan N lebih besar dari nol. jika benar dilanjutkan dengan memeriksa bilangan genap atau ganjil dengan if N % 2 == 0 digunakan untuk memeriksa apakah bilangan N habis dibagi 2, atau genap. Jika ini benar,bilangan tersebut adalah bilangan positif genap.Jika tidak habis dibagi 2 (kondisi else pada baris berikutnya), maka program menampilkan bahwa bilangan tersebut positif ganjil.

Jika kondisi N > 0 tidak terpenuhi, maka kode elif N < 0 akan dieksekusi, yaitu ketika N lebih kecil dari nol maka bilangan negatif.Jika kedua kondisi tersebut salah (berarti N sama dengan nol), program akan menampilkan bahwa N adalah nol.

- Interpretasi Hasil:

Masukkan nilai N: 2

2 adalah bilangan positif genap

Problem (2) Kalkulator Sederhana

In [ ]:
# Meminta input angka dan operator dari pengguna
angka1 = int(input("Masukkan angka pertama: "))
angka2 = int(input("Masukkan angka kedua: "))
operator = input("Masukkan operator (+, -, *, /, %): ")

# Memeriksa operator dan melakukan perhitungan
if operator == '+':
    hasil = angka1 + angka2
    print(f"{angka1} + {angka2} = {hasil}")
elif operator == '-':
    hasil = angka1 - angka2
    print(f"{angka1} - {angka2} = {hasil}")
elif operator == '*':
    hasil = angka1 * angka2
    print(f"{angka1} * {angka2} = {hasil}")
elif operator == '/':
    hasil = angka1 // angka2  # dibulatkan ke bawah
    print(f"{angka1} / {angka2} = {hasil}")
elif operator == '%':
    hasil = angka1 % angka2
    print(f"{angka1} % {angka2} = {hasil}")
else:
    print("Operator tidak valid.")

Masukkan angka pertama: 2
Masukkan angka kedua: 6
Masukkan operator (+, -, *, /, %): +
2 + 6 = 8


In [ ]:
# Meminta input angka dan operator dari pengguna
angka1 = int(input("Masukkan angka pertama: "))
angka2 = int(input("Masukkan angka kedua: "))
operator = input("Masukkan operator (+, -, *, /, %): ")

# Memeriksa operator dan melakukan perhitungan
if operator == '+':
    hasil = angka1 + angka2
    print(f"{angka1} + {angka2} = {hasil}")
elif operator == '-':
    hasil = angka1 - angka2
    print(f"{angka1} - {angka2} = {hasil}")
elif operator == '*':
    hasil = angka1 * angka2
    print(f"{angka1} * {angka2} = {hasil}")
elif operator == '/':
    hasil = angka1 // angka2  # dibulatkan ke bawah
    print(f"{angka1} / {angka2} = {hasil}")
elif operator == '%':
    hasil = angka1 % angka2
    print(f"{angka1} % {angka2} = {hasil}")
else:
    print("Operator tidak valid.")

Masukkan angka pertama: 13
Masukkan angka kedua: 5
Masukkan operator (+, -, *, /, %): %
13 % 5 = 3


- Interpretasi coding:

Meminta masukkan dua angka,angka prtama dan kedua. memilih operator mana yang akan digunakan kemudia membuat operasi tersebut dan mengeluarkan output yang sesuai. Tetapijika operator yang digunakan tidak sesuai maka akan muncul "operator tidak valid"

- Interpretasi Hasil:

contoh 1:

Masukkan angka pertama: 2

Masukkan angka kedua: 6

Masukkan operator (+, -, *, /, %): +

2 + 6 = 8

contoh 2 :

Masukkan angka pertama: 13

Masukkan angka kedua: 5

Masukkan operator (+, -, *, /, %): %

13 % 5 = 3

#🏆 POST TEST 🎯

Problem (1)

In [ ]:
# Input harga dasar dan harga jual untuk setiap barang
harga_dasar_a = int(input("Masukkan harga dasar barang A: "))
harga_jual_a = int(input("Masukkan harga jual barang A: "))
harga_dasar_b = int(input("Masukkan harga dasar barang B: "))
harga_jual_b = int(input("Masukkan harga jual barang B: "))
harga_dasar_c = int(input("Masukkan harga dasar barang C: "))
harga_jual_c = int(input("Masukkan harga jual barang C: "))

# Menghitung keuntungan untuk setiap barang
keuntungan_a = harga_jual_a - harga_dasar_a
keuntungan_b = harga_jual_b - harga_dasar_b
keuntungan_c = harga_jual_c - harga_dasar_c

# Menentukan barang dengan keuntungan terbesar
if keuntungan_a > keuntungan_b and keuntungan_a > keuntungan_c:
  barang_terbaik = "A"
elif keuntungan_b > keuntungan_a and keuntungan_b > keuntungan_c:
  barang_terbaik = "B"
else:
  barang_terbaik = "C"

# Menampilkan barang yang harus ditawarkan
print(f"Barang yang harus ditawarkan adalah barang {barang_terbaik}")

Masukkan harga dasar barang A: 8
Masukkan harga jual barang A: 11
Masukkan harga dasar barang B: 7
Masukkan harga jual barang B: 12
Masukkan harga dasar barang C: 6
Masukkan harga jual barang C: 12
Barang yang harus ditawarkan adalah barang C


- Interpretasi coding:

Meminta pengguna memasukkan harga dasar dan harga jual dari tiga barang (A,B, dan C).kemudian menghitung keuntungan dengan mengurangkan harga dasar dengan harga jualnya.
Struktur if-elif-else digunakan untuk memeriksa mana di antara keuntungan_a, keuntungan_b, atau keuntungan_c yang paling besar:

	•	Jika keuntungan_a terbesar, variabel barang_terbaik diset menjadi "A".
	•	Jika keuntungan_b terbesar, variabel barang_terbaik diset menjadi "B".
	•	Jika tidak ada yang lebih besar dari keuntungan_c, maka barang_terbaik diset menjadi "C".
Menampilkan Barang dengan Keuntungan Terbesar

- Interpretasi hasil:

Test Case 1 :

Masukkan harga dasar barang A: 10

Masukkan harga jual barang A: 15

Masukkan harga dasar barang B: 10

Masukkan harga jual barang B: 12

Masukkan harga dasar barang C: 10

Masukkan harga jual barang C: 11

Barang yang harus ditawarkan adalah barang A

Test case 2 :

Masukkan harga dasar barang A: 8

Masukkan harga jual barang A: 11

Masukkan harga dasar barang B: 7

Masukkan harga jual barang B: 12

Masukkan harga dasar barang C: 6

Masukkan harga jual barang C: 12

Barang yang harus ditawarkan adalah barang C

Problem (2)

In [ ]:
def kelas_mahasiswa(nrp_akhir):
    nrp_akhir = int(nrp_akhir)

    # menentukan kelas berdasarkan rentang NRP
    if nrp_akhir > 300:
        kelas = "K7" if nrp_akhir % 2 != 0 else "K8"
    elif nrp_akhir > 200:
        kelas = "K5" if nrp_akhir % 2 != 0 else "K6"
    elif nrp_akhir > 100:
        kelas = "K3" if nrp_akhir % 2 != 0 else "K4"
    elif nrp_akhir > 0:
        kelas = "K1" if nrp_akhir % 2 != 0 else "K2"
    else:
        kelas = "NRP Tidak Valid"

    return kelas

# input 3 digit terakhir NRP mahasiswa
nrp = input("Masukkan 3 digit terakhir NRP: ")
kelas = kelas_mahasiswa(nrp)

# menampilkan hasil
print(f"Mahasiswa dengan NRP {nrp} berada di kelas {kelas}.")

Masukkan 3 digit terakhir NRP: 121
Mahasiswa dengan NRP 121 berada di kelas K3.


In [ ]:
def kelas_mahasiswa(nrp_akhir):
    nrp_akhir = int(nrp_akhir)

    # menentukan kelas berdasarkan rentang NRP
    if nrp_akhir > 300:
        kelas = "K7" if nrp_akhir % 2 != 0 else "K8"
    elif nrp_akhir > 200:
        kelas = "K5" if nrp_akhir % 2 != 0 else "K6"
    elif nrp_akhir > 100:
        kelas = "K3" if nrp_akhir % 2 != 0 else "K4"
    elif nrp_akhir > 0:
        kelas = "K1" if nrp_akhir % 2 != 0 else "K2"
    else:
        kelas = "NRP Tidak Valid"

    return kelas

# input 3 digit terakhir NRP mahasiswa
nrp = input("Masukkan 3 digit terakhir NRP: ")
kelas = kelas_mahasiswa(nrp)

# menampilkan hasil
print(f"Mahasiswa dengan NRP {nrp} berada di kelas {kelas}.")

Masukkan 3 digit terakhir NRP: 544
Mahasiswa dengan NRP 544 berada di kelas K8.


Interpretasi Coding :

1.	Fungsi kelas_mahasiswa:
	•	Menerima input tiga digit terakhir NRP (nrp_akhir) dan mengonversinya menjadi bilangan bulat.

	•	Menggunakan kondisi if-elif untuk menentukan kelas:

	•	Jika nrp_akhir > 300, kelasnya K7 (ganjil) atau K8 (genap).

	•	Jika nrp_akhir > 200, kelasnya K5 (ganjil) atau K6 (genap).

	•	Jika nrp_akhir > 100, kelasnya K3 (ganjil) atau K4 (genap).

	•	Jika nrp_akhir > 0, kelasnya K1 (ganjil) atau K2 (genap).

	•	Jika nrp_akhir ≤ 0, hasilnya “NRP Tidak Valid.”

	•	Fungsi mengembalikan kelas yang sesuai.

	2.	Penggunaan Fungsi:

	•	Program meminta tiga digit terakhir NRP dari pengguna, memanggil kelas_mahasiswa, dan menampilkan kelas berdasarkan NRP tersebut.

  Interpretasi hasil :

  - Test case 1 :

Masukkan 3 digit terakhir NRP: 121

Mahasiswa dengan NRP 121 berada di kelas K3.

- Test case 2 :

Masukkan 3 digit terakhir NRP: 544

Mahasiswa dengan NRP 544 berada di kelas K8.

📌Catatan:
  - Notebook python ini dibuat untuk memudahkan temen-temen belajar ngoding dan pengumpulan tugas.
  - Silakan download notebook ini lalu upload di Google Collab milik kalian agar format `pertemuan`, `topik pertemuan`,
`tanggal`, `deskripsi pembahasan materi`, `baris coding` dan `interpretasi` bisa disesuaikan dengan kebutuhan teman-teman.

  - Interpretasi coding dan hasil dimaksudkan agar teman-teman semakin memahami apa yang dipelajari.

  - Deskripsi pembahasan materi dimaksudkan agar teman-teman semakin memahami apa fokus yang dipelajari.

  - Tulisan di setiap sebelum baris code disebut `markdown`. Ada beberapa jenis simbol yang memudahkan kepenulisan `markdown` kita, yaitu seperti : untuk mengatur ukuran tulisan biasanya tulisan pada header menggunakan ini`#`, `##`, `###`, untuk membuat pointer menggunakan ini `-`, untuk membuat tulisan ter-highlight menggunakan ini *` `* , untuk short-cut membuat tulisan jadi **bold** gunakan ini `** **`, dan untuk *tulisan miring* gunakan ini `* *` .

  - Semangat belajar! 🌟✨

©️ LABORATORIUM SAINS DATA 2024